app/models/miq_worker/runner.rb

```diff
@@ -341,6 +341,10 @@ def do_work_loop
         @backoff = nil
       end
 
+      # Should be caught by the rescue in `#start` and will run do_exit from
+      # there.
+      raise Interrupt if @sigterm_recieved
```

Comment:
@jrafanie good catch on both of those.  Accessing a instance var will cause warnings when in verbose mode (so these probably exist in in the test output from travis), so yeah, those should be fixed.

Will open up a follow up PR shortly.